commons/src/main/java/com/jagrosh/jdautilities/commons/waiter/EventWaiter.java

```diff
@@ -45,26 +47,44 @@
  * <br>A more "shutdown adaptable" constructor allows the provision of a
  * {@code ScheduledExecutorService} and a choice of how exactly shutdown will be handled
  * (see {@link EventWaiter#EventWaiter(ScheduledExecutorService, boolean)} for more details).
- * 
+ *
+ * <p>By default, waiting events are stored in a {@link java.util.HashMap HashMap}, and thread
+ * safety is not considered.
+ * <br>If you require a thread safe EventWaiter that allows concurrent modification of the
+ * stored events, use {@link EventWaiter#EventWaiter(ScheduledExecutorService, boolean, boolean)}.
+ *
  * <p>As a final note, if you intend to use the EventWaiter, it is highly recommended you <b>DO NOT</b>
  * create multiple EventWaiters! Doing this will cause unnecessary increases in memory usage.
- * 
+ *
  * @author John Grosh (jagrosh)
  */
-public class EventWaiter implements EventListener
+public class EventWaiter implements IEventWaiter, EventListener
 {
-    private final HashMap<Class<?>, Set<WaitingEvent>> waitingEvents;
+    private final Map<Class<?>, Set<WaitingEvent>> waitingEvents;
     private final ScheduledExecutorService threadpool;
     private final boolean shutdownAutomatically;
-    
+    private final boolean threadSafe;
+
     /**
-     * Constructs an empty EventWaiter.
+     * Constructs a default EventWaiter.
      */
     public EventWaiter()
     {
         this(Executors.newSingleThreadScheduledExecutor(), true);
     }
 
+    /**
+     * Creates a new EventWaiter that is thread safe allowing safe modification
+     * across multiple threads when {@code threadSafe} is {@code true}.
+     *
+     * @param  threadSafe
+     *         {@code true} if the EventWaiter constructed is thread-safe.
+     */
+    public EventWaiter(boolean threadSafe)
+    {
+        this(Executors.newSingleThreadScheduledExecutor(), true, true);
```

Comment:
Not using the local `threadSafe` for calling the overloaded constructor.
```suggestion
        this(Executors.newSingleThreadScheduledExecutor(), true, threadSafe);
```